src/cogent3/core/new_alignment.py

```diff
@@ -297,25 +310,62 @@ def __init__(
         *,
         seqs_data: SeqsData,
         moltype: new_moltype.MolType,
+        names: list[str] = None,
         info: Union[dict, InfoClass] = None,
     ):
         self._seqs_data = seqs_data
         self.moltype = moltype
-        self._names = self._seqs_data.names
+        self.names = names
         self._seqs_data.make_seq = self.moltype.make_seq
         if not isinstance(info, InfoClass):
             info = InfoClass(info) if info else InfoClass()
         self.info = info
         self._repr_policy = dict(num_seqs=10, num_pos=60, ref_name="longest", wrap=60)
+        self._annotation_db = None
 
     @property
     def names(self) -> Iterator[str]:
         return self._names
 
+    @names.setter
+    def names(self, names: list[str]):
```

Comment:
add a `# refactor: design` comment (or something better if you can think of it). I'm not sure if this will be a good idea and I know I suggested it, but definitely worth trialing.